### PR TITLE
Add "shadowable" check; rename jf

### PR DESCRIPTION
**Pull request by ryanlaws**
Add get_shadowable_attribute, which checks whether the track is a
shadow, removing some duplication. Use this for is_jf, which has been
renamed. This could be used for other shadowable attributes.

### CODE REVIEW

File: lib/Track.lua [hunk 1]
```diff
@@ -42,7 +42,7 @@ function Track:new(x)
   t.device = 1
   -- crow
   t.pair = 1
-  t.jf = false
+  t.is_jf = false
```
Comment (tyleretters) [inline lib/Track.lua]: the `is_` prefix is reserved for methods. the attributes are always whatever follows the underscore.

Comment (tyleretters) [reply lib/Track.lua]: please change back to `t.jf`

File: lib/Track.lua [hunk 2]
```diff
@@ -274,10 +274,18 @@ function Track:get_shadow_attribute(attribute)
   elseif attribute == "channel" then return track:get_channel()
   elseif attribute == "device"  then return track:get_device()
   elseif attribute == "pair"    then return track:get_pair()
-  elseif attribute == "jf"      then return track:is_jf()
+  elseif attribute == "is_jf"   then return track:get_is_jf()
```
Comment (tyleretters) [inline lib/Track.lua]: similar to the `t.jf` comment - `is_` is the convention for boolean methods. the `get_` is not used with boolean getters.

File: lib/Track.lua [hunk 3]
```diff
@@ -274,10 +274,18 @@ function Track:get_shadow_attribute(attribute)
...
   end
 end
 
+function Track:get_shadowable_attribute(attribute)
```
Comment (tyleretters) [inline lib/Track.lua:281]: this is good. i'll need to test but i think this will help with some of the issues.

**Review by tyleretters — CHANGES_REQUESTED** [review]

File: lib/Track.lua (hunk 1 above)
Comment (ryanlaws) [reply lib/Track.lua]: Changed back and modified `get_shadowable_attribute` check - see 2c04cd3

File: lib/Track.lua (hunk 2 above)
Comment (ryanlaws) [reply lib/Track.lua]: Changed back and modified get_shadowable_attribute check - see 2c04cd3

File: lib/Track.lua (hunk 3 above)
Comment (ryanlaws) [reply lib/Track.lua:281]: Sounds good!

**Review by tyleretters — APPROVED** [review]

**Comment by tyleretters** [comment]
nice!!!

**Pull request merged**